webapp/app/routes.py

```diff
@@ -115,7 +115,9 @@ def add_http_header(response):
 
     @app.route('/eligibility/step/<step>', methods=['GET', 'POST'])
     def eligibility(step):
-        return EligibilityStepChooser(endpoint='eligibility').get_correct_step(step_name=step).handle()
+        return EligibilityStepChooser(endpoint='eligibility') \
+            .get_correct_step(step_name=step, update_data=request.method == 'POST') \
+            .handle()
```

Comment:
I would factor out a variable here, it's easy to miss the logic otherwise.

```suggestion
        update_data = request.method == 'POST'
        return EligibilityStepChooser(endpoint='eligibility') \
            .get_correct_step(step_name=step, update_data=update_data) \
            .handle()
```